.github/workflows/coverage_report.yml

```diff
@@ -0,0 +1,62 @@
+name: Add coverage comment on PR
+
+on:
+  workflow_run:
+    workflows: ["Test and lint"]
+    types:
+      - completed
+
+permissions: {}
+
+jobs:
+  cov-report:
+    permissions:
+        contents: read
+        pull-requests: write
+    runs-on: ubuntu-latest
+    if: >
+      github.event.workflow_run.event == 'pull_request'
+      && github.repository == github.event.workflow_run.repository.full_name
+    # The github.repository == ... check will prevent this workflow from running on forks.
+    steps:
+      - name: Get PR number
+        id: get_pr_number
+        env:
+          SHA: "${{ github.event.workflow_run.head_sha }}"
```

Comment:
Do you have any docs for this field? I couldn't find anything that clearly documents that it would point to the `HEAD` to the PR's branch, rather than the `HEAD` of the main branch.

(It's also not clear to me how this works when PRs contain some shared history or have identical commits due to cherry picking, so this feels pretty risky.)